Feat/#53 고민작성 디자인 변경 및 버그 수정

IslandofDream opened the pull request:
<!-- 🔥 Assignee, Label, Reviewer 설정!!!🔥 -->

## 📒 이슈

- Resolved: #53

## 🔥 작업 내용

- 수현이가 보내준 폰트, 앱 백그라운드 들어가면 초기화, 저장 다이얼로그 누락등등 수정했습니다.
- 시하가 바꿔준 디자인안대로 일부 속성 수정하였습니다. 

## ☑️ PR Point

<!-- 피드백을 받고 싶은 부분, 공유하고 싶은 부분, 작업 과정, 이유를 적어주세요. -->

- 앱 백그라운드 들아갔다 다시 들어가면 텍스트 초기화되는 거는 stateFlow를 다시 collect해서 render함수를 재실행해서 생겼음
그런데 stateFlow는 중복값은 다시 방출을 안할텐대 왜 다시 수집하는지는 의문..
- onStop에서 백그라운드 전환전 현재 id를 저장해 render함수가 재 호출되어 텍스트 초기화 되는걸 방지
- 이외 자잘자잘한 부분들 수정

## 📱실행결과
- 다이얼로그 생성시 약간 어색했던 부분은 애니메이션 적용으로 해결 

https://github.com/TeamHARA/KAERA_Android/assets/70648111/3e65f8ea-e7ee-4ca3-8336-c7484dc7b125


<!-- gif or mp4. gif는 [https://ezgif.com/](https://ezgif.com/) 활용! 용량제한 10MB넘어가면 카톡으로.. -->

IslandofDream committed:
[Refactor] EditText 누락폰트 적용

IslandofDream committed:
[Feat] 앱 백그라운드 적용시 텍스트가 사라지던 현상 수정

IslandofDream committed:
[Feat] 기록저장 다이얼로그 조건 누락 수정

IslandofDream committed:
[Feat] 다이얼로그 팝업 애니메이션 추가

IslandofDream committed:
[Refactor] 고민작성 디자인 변경안 적용

IslandofDream commented:
애니메이션만 시하가 알려주면 머지할께요

wkdyujin commented on app/src/main/java/com/hara/kaera/presentation/write/viewmodel/WriteViewModel.kt:
```diff
@@ -16,34 +17,51 @@ class WriteViewModel @Inject constructor(
     fun getTemplateDetailData() {
         _templateDetailFlow.value = UiState.Loading
-        viewModelScope.launch {
+        viewModelScope.launch(Dispatchers.IO) {
```
왜 여기서 코루틴을 디폴트로 안 보내고 IO디스패처로 보내나요??

IslandofDream replied:
https://thdev.tech/kotlin/2021/01/12/Retrofit-Coroutines/
IO 작업이라 해줬는데 좀 찾아보니까 할필요 없다네 ㅋㅋ

wkdyujin commented on app/src/main/java/com/hara/kaera/presentation/write/viewmodel/WriteViewModel.kt:
```diff
@@ -16,34 +17,51 @@ class WriteViewModel @Inject constructor(
 
+    private val _curTemplateIdFlow = MutableStateFlow(PrevId.INIT.value)
+    val curTemplateIdFlow = _curTemplateIdFlow.asStateFlow()
+    //repeatOnLifeCycle에 따라서 백그라운드로 나갔다 다시 create되면
+    //onStarted에서 tmplateIdFlow를 다시 collect함 그에 따라서
+    // getTemplateDetailData함수를 재호출함
+    // 그에따라 새로 render함수도 호출 되고 이전에 작성한 글이 render가 다시 되어서 작성했던 글들이 삭제되어 버림
+    // 이를 방지하기 위해서 나갔다 들어와도 재호출되지 않도록 singleSelecteAdapter 처럼 이전 선택한
+    // id를 저장해두었다가 조건검사에 써준다
```
원인 어떻게 이렇게 잘 찾았어 대단하다

skylartosf replied:
진짜 신기해

wkdyujin commented on app/src/main/java/com/hara/kaera/presentation/write/viewmodel/WriteViewModel.kt:
```diff
@@ -16,34 +17,51 @@ class WriteViewModel @Inject constructor(
         }
     }
 
+    enum class PrevId(val value : Int) {
+        INIT(-1)
+    }
```
상수는 companion object 쓰는 줄 알았는데 enum이네! 비슷한 역할이라 코드 스타일 차이인가?

IslandofDream replied:
상수 관리할때는 Enum 이 맞는거 같고 companion object는 외부에서의 접근을 도와주는 자바로 치면 static의 느낌!

wkdyujin reviewed (approved):
짱입니다😆 고마워

IslandofDream committed:
[Refactor] 상수 네이밍 변경

IslandofDream commented:
## 09/08
- 다이얼로그는 페이드인 페이드 아웃 애니메이션 xml 적용시켜서 BindingDialogFragment에 적용예정!

IslandofDream committed:
[Refactor] 커스텀 스낵바 색상 오적용 변경

IslandofDream committed:
[Refactor] 애니메이션 페이드 인 아웃으로 변경 및 어색한 다이얼로그 애니메이션 수정
- <item name="android:windowIsFloating">false</item>
얘 때문에 다이얼로그가 나타날때 어색한 애니메이션이 나왔음

skylartosf reviewed (approved):
고생하셨습니다 💖
